source/client.c

```diff
@@ -1800,37 +1929,53 @@ uint16_t aws_mqtt_client_connection_subscribe_local(
 static bool s_reconnect_resub_iterator(const struct aws_byte_cursor *topic, enum aws_mqtt_qos qos, void *user_data) {
     struct subscribe_task_arg *task_arg = user_data;
 
+    struct subscribe_task_topic *task_topic =
+        aws_mem_calloc(task_arg->connection->allocator, 1, sizeof(struct subscribe_task_topic));
     struct aws_mqtt_topic_subscription sub;
     AWS_ZERO_STRUCT(sub);
     sub.topic = *topic;
     sub.qos = qos;
+    task_topic->request = sub;
 
-    aws_array_list_push_back(&task_arg->topics, &sub);
+    aws_array_list_push_back(&task_arg->topics, &task_topic);
 
     return true;
 }
 
 static enum aws_mqtt_client_request_state s_resubscribe_send(
-    uint16_t message_id,
+    uint16_t packet_id,
     bool is_first_attempt,
     void *userdata) {
 
-    (void)is_first_attempt;
-
     struct subscribe_task_arg *task_arg = userdata;
     bool initing_packet = task_arg->subscribe.fixed_header.packet_type == 0;
     struct aws_io_message *message = NULL;
 
+    size_t sub_count = aws_mqtt_topic_tree_get_sub_count(&task_arg->connection->thread_data.subscriptions);
+    if (sub_count == 0) {
+        aws_raise_error(AWS_ERROR_MQTT_NO_TOPICS_FOR_RESUBSCRIBE);
+        AWS_LOGF_WARN(
+            AWS_LS_MQTT_CLIENT,
+            "id=%p: Not subscribed to any topics. Resubscribe is unnecessary, no packet will be sent. Error %s.",
```

Comment:
How would a user have caused this? Is there something else we should tell them, or should we just silently ignore this situation?

Reply:
The previous connection had no subscription, then the resubscribe called. Yeah, it's probably better to silently ignore this